docs/source/configuration/authn-jwt.mdx

```diff
@@ -1,107 +1,129 @@
- - The key retrieval cooldown duration is customized: 30 seconds
+<tbody>
+<tr class="required">
+<td style="min-width: 150px;">
 
-```yaml title="complete.yaml"
-authentication:
-  experimental:
-    jwt:
-      jwks_url: https://dev-zzp5enui.us.auth0.com/.well-known/jwks.json
-      header_name: Different
-      header_value_prefix: Prefix
-      cooldown: 30s
-```
+##### `jwks_url`
 
-There are some considerations for each of these parameters.
+</td>
+<td>
 
-### jwks_url
+**Required.** The URL of the JWK Set (JWKS) that the router should use. Can be an `https://` or a `file://` URL.
 
-If you are using a symmetric algorithm (e.g.: HS256) **DO NOT** place your JWKS on the network. They should only be loaded from disk.
+**If _any_ of your JWKs uses a symmetric signature algorithm (such as `HS256`), always use a `file://` URL.** Symmetric signature algorithms use a shared key that should never be accessible over the network.
 
-### header_name and header_value_prefix
+</td>
+</tr>
 
-Your chosen header must be a valid HTTP header name and your prefix should not contain any white space.
+<tr>
+<td style="min-width: 150px;">
 
-### cooldown
+##### `header_name`
 
-The default value for this parameter, 15s, should be suitable for almost all use cases. However, it is provided so that advanced users can configure this if required.
+</td>
+<td>
 
-## Observability
+The name of the HTTP header that client requests will use to provide their JWT to the router. Must be a valid name for an HTTP header.
 
-### Tracing
+The default value is `Authorization`.
 
-There is a tracing span for the authentication plugin. The span name is "authentication plugin".
+</td>
+</tr>
 
-### Metrics
+<tr>
+<td style="min-width: 150px;">
 
-Three metrics are provided and exported via prometheus. They look like this:
+##### `header_value_prefix`
 
-```
-# HELP apollo_authentication_failure_count apollo_authentication_failure_count
-# TYPE apollo_authentication_failure_count counter
-apollo_authentication_failure_count{kind="JWT",service_name="apollo-router"} 1
-# HELP apollo_authentication_success_count apollo_authentication_success_count
-# TYPE apollo_authentication_success_count counter
-apollo_authentication_success_count{kind="JWT",service_name="apollo-router"} 11
-# HELP apollo_authentication_cooldown_count apollo_authentication_cooldown_count
-# TYPE apollo_authentication_cooldown_count counter
-apollo_authentication_cooldown_count{kind="JWT",service_name="apollo-router"} 3
-```
+</td>
+<td>
+
+The string that will always precede the JWT in the header value corresponding to [`header_name`](#header_name). This value must not include whitespace.
+
+The default value is `Bearer`.
+
+</td>
+</tr>
+
+<tr>
+<td style="min-width: 150px;">
+
+##### `cooldown`
+
+</td>
+<td>
 
-## Validation
+The frequency at which the router fetches an up-to-date JWK Set from [`jwks_url`](#jwks_url). In almost all cases, you don't need to set a custom value for this option.
```

Comment:
This is not how cooldown works. It isn't a polling interval, it's an amount of time for which the router will not allow the appearance of a new "kid" to trigger a re-fetch of the JWKS.

Not sure how to phrase that better, but I think your current phrasing needs to be re-worked.

Reply:
@StephenBarlow `kid` = Key Id.  There's some clarification to it in https://www.rfc-editor.org/rfc/rfc7517.html#section-4.5

@garypen Do you think we need to introduce the concept of Key Id to this document to help explain how this option works more?  Also, were other names considered for `cooldown`?  I don't see it come up in a Google search for "JWKS cooldown" so I'm not sure if this is common convention or not.

Reply:
I think it would be helpful to specify what a "kid" is.

Regarding cooldown, I'm open to alternative names. I don't think you would find "JWKS cooldown" because I believe most approaches are probably polling. I did a quick google search and couldn't find anything useful.

Maybe something like: "minimum-duration-between-kid-initiated-jwks-update", but snappier?

Reply:
Yeah, like, `key_id_update_throttle_duration` or something, but I'm also not finding the snappiest of snappy names yet in my 🧠

Reply:
Thanks! This was definitely a case of "I'll write a guess and find out how wrong I am 😄 "

Reply:
@garypen I think I've documented this correctly now via the changes in https://github.com/apollographql/router/pull/2476/commits/89261280d380c4b6c4a090e159190ad760c2d6bd

Could you verify whether everything in that commit looks all right? Hopefully can merge this one soon for more tactical additions to follow